### ISSUE TITLE
Getting error "the trait bound `(): Tokenizable` is not satisfied" when calling contract.query()

### ISSUE THREAD

**Issue by eute** [tomusdrw/rust-web3#506]
I'm new to the rust-web3 crate, so forgive me for any potential beginner mistakes.
When I try to query the balance for an ERC-20 token, I get this compile error:

```rust
error[E0277]: the trait bound `(): Tokenizable` is not satisfied
   --> src\main.rs:242:29
    |
242 |                     let balance = contract.query("balanceOf", (public_key,), None, web3::contract::Options::default(), None).await....
    |                                            ^^^^^ the trait `Tokenizable` is not implemented for `()`
    |
    = note: required because of the requirements on the impl of `Detokenize` for `()`
```

I followed the example `examples/contract.rs`, specifically the section under `// Accessing existing contract` but to no avail. Regardless, it doesn't even seem to be affected by anything I pass into it or change around it, it's like the function just doesn't work outright. Any idea what could be causing this?

**Comment by eute** [comment]
Okay, I checked the source code of `query()`, noticed that the variable with the `Detokenize` trait is the return variable, and now I seem to have fixed it by stating the return type upfront like so:
```rust
contract.query::<U256, _, _, _>("balanceOf", (public_key,), None, web3::contract::Options::default(), None)
```
Noticed that `examples/contract.rs` does this by declaring the type of the `balance` variable, which is also a solution. My mistake for missing that, although could this be documented and warned about in a more explicit manner? Neither the docs nor the example point that out. In my opinion it's hard to figure out such a critical detail on your own if you have missed it, especially when you get a vague compile error like that.

**Comment by tomusdrw** [comment]
Good point. Addressed in #507
